### PR TITLE
211 add chain id to package state

### LINKED ISSUE #211: Add `chainId` to package states when building
For example, on this [`deployInfo`](https://ipfs.usecannon.com/Qmer9CWSbD1MtTDhj3yUCNZGpizaPjUAsqC6t9C2rmqDhA?compression=true&format=json) should be available as `deployInfo.chainId`.

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
resolve #211

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -66,3 +66,6 @@ typings/
 
 # macOS
 .DS_Store
+
+
```
[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply .gitignore]: Thanks, Done.

**Comment by mohammadranjbarz** [comment]
> Hey @mohammadranjbarz ! looks good, it is just also needed when executing `provision` steps during builds, here: https://github.com/usecannon/cannon/blob/211_add_chainId_to_package_state/packages/builder/src/steps/provision.ts/#L188

@mjlescano  Thanks for mentioning, I added it

**Review by dbeal-eth — APPROVED** [review]

**Comment by dbeal-eth** [comment]
@mohammadranjbarz did you test running a build with this updated code and verify that the `chainId` is in fact included in the resulting cannon package on IPFS? just want to make sure we are in the habit of testing our changes in practice, at a minimum, before merging.

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz did you test running a build with this updated code and verify that the `chainId` is in fact included in the resulting cannon package on IPFS? just want to make sure we are in the habit of testing our changes in practice, at a minimum, before merging.

No I couldn't run it, because it uses typings imported from `@usecannon/builder` so although I changed typing of `DeploymentInfo` and added `chainId` but as it has not been published , my code tells you can pass `chainId` I thought we should merge it and after `@usecannon/builder` got published (assumed merging branch publish that package), we can test it

**Comment by aminlatifi** [comment]
@mohammadranjbarz running `npm run build` on the `build` package will update compiled version which is imported by other packages. I have the env set up already on my machine, just please merge latest `main` branch  into yours

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz running `npm run build` on the `build` package will update compiled version which is imported by other packages. I have the env set up already on my machine, just please merge latest `main` branch into yours

Thanks for your feedback, done

**Comment by aminlatifi** [comment]
I was curious whether it works, and found another bug! It doesn't work since the build process has a bug in publishing!
The URL is used for publishing to the IPFS is the one filled in the "ipfsUrl", not "publishIpfsUrl"

**Comment by aminlatifi** [comment]
> I was curious whether it works, and found another bug! It doesn't work since the build process has a bug in publishing! The URL is used for publishing to the IPFS is the one filled in the "ipfsUrl", not "publishIpfsUrl"

Reported here
https://github.com/usecannon/cannon/issues/225

**Comment by aminlatifi** [comment]
Tested, chainId was there :)

**Pull request merged**